akka-stream/src/main/scala/akka/stream/javadsl/Sink.scala

```diff
@@ -308,6 +308,26 @@ object Sink {
     new Sink(scaladsl.Sink.combine(output1.asScala, output2.asScala, seq: _*)(num => strategy.apply(num)))
   }
 
+  /**
+   * Creates a `Sink` that is materialized as an [[akka.stream.javadsl.SinkQueueWithCancel]].
+   * [[akka.stream.javadsl.SinkQueueWithCancel.pull]] method is pulling element from the stream and returns ``CompletionStage[Option[T]]``.
+   * `CompletionStage` completes when element is available.
+   *
+   * Before calling pull method second time need to ensure that request buffer is not full or wait until some of the previous CompletionStages completes.
+   * Pull returns Failed future with ''IllegalStateException'' if request buffer is full.
+   *
+   * `Sink` will request at most number of elements equal to size of `inputBuffer` from
+   * upstream and then stop back pressure.  You can configure size of input
+   * buffer by using [[Sink.withAttributes]] method.
```

Comment:
Hmm I wonder why it should have a buffer at all, you can always compose with a buffer upstream. Would simplify the implementation a lot.

Reply:
Totally agree, left only for backward compatibility.

Reply:
I've created #28270 to track that, no need to change here.